lib/codecs/src/encoding/format/csv.rs

```diff
@@ -85,7 +86,9 @@ impl Encoder<Event> for CsvSerializer {
                 Some(Value::Integer(int)) => wtr.write_field(int.to_string())?,
                 Some(Value::Float(float)) => wtr.write_field(float.to_string())?,
                 Some(Value::Boolean(bool)) => wtr.write_field(bool.to_string())?,
-                Some(Value::Timestamp(timestamp)) => wtr.write_field(timestamp.to_rfc3339())?,
+                Some(Value::Timestamp(timestamp)) => {
+                    wtr.write_field(timestamp.to_rfc3339_opts(SecondsFormat::AutoSi, true))?
+                }
```

Comment:
`Value::Timestamp`'s Display implementation can be found [here](https://github.com/vectordotdev/vector/blob/master/lib/vrl/value/src/value/display.rs#L38).